### PR TITLE
Hackday-testing and additions

### PR DESCRIPTION
**Pull request by alexthemayers**

### CODE REVIEW

File: openapi/media_type.go [hunk 1]
```diff
@@ -2,26 +2,32 @@ package openapi
...
 
-func (m *MediaType) GetName() string {
-	return m.parent.GetName() + m.name
-}
-
 func (m *MediaType) getRef() string {
 	return ""
 }
 
+// GetName overrides the default node implementation to avoid
```
Comment (SasSwart) [inline openapi/media_type.go]: Incomplete comment here. What were you going to say?

File: openapi/media_type_test.go [hunk 2]
```diff
@@ -5,57 +5,47 @@ import (
...
-		t.Logf("MediaType.getChildren() returned a nil schema")
-		t.Fail()
+	openapi, _ := LoadOpenAPI(absOpenAPI)
+	mt := Dig(openapi, testEndpoint, testMethod, testReqBody, testMediaType)
+	children := mt.getChildren()
+	for model, schema := range children {
+		if model == testSchema {
+			s, ok := schema.(*Schema)
+			if !ok {
+				t.Errorf("MediaType.getChildren() didn't return a *Schema")
```
Comment (SasSwart) [inline openapi/media_type_test.go:27]: nonblocking: These tests would be more valuable if we asserted against the content of the schema object we expect to make sure we get the right one. I'll consider sprucing it up at some point. This is already progress

File: openapi/node.go [hunk 3]
```diff
@@ -64,7 +64,8 @@ func (n *node) GetOutputFile() string {
 }
 
 func (n *node) getRef() string {
-	return n.ref
+	panic("(n *node) getRef(): We don't call this one")
```
Comment (SasSwart) [inline openapi/node.go]: please keep this message consistent with the ones in setChild and SetMetadata. I like the `(n *node) getRef():` part, but `not implemented by composed type` is more clear than `We don't call this one`.

Consider a constant message at the top of the file to return in these methods.

File: openapi/node_test.go [hunk 4]
```diff
@@ -0,0 +1,31 @@
+package openapi
+
+import "testing"
+
+func TestOpenAPI_Dig(t *testing.T) {
+	openapi, _ := LoadOpenAPI(absOpenAPI)
+	endpoint := Dig(openapi, testEndpoint)
+	if _, ok := endpoint.(*PathItem); !ok {
```
Comment (SasSwart) [inline openapi/node_test.go:9]: Non blocking: These tests would be more useful if they checked for identity, not just type.

File: openapi/openapi.go [hunk 5]
```diff
@@ -11,11 +11,27 @@ type Config struct {
 	OpenAPIFile string
 }
 
+// OpenAPI is a programmatic representation of the OpenApi Document object defined here: https://swagger.io/specification/#openapi-object
+type OpenAPI struct {
+	node
+	OpenAPI string `yaml:"openapi"`
+	Info    Info
+	//Servers Servers
```
Comment (SasSwart) [inline openapi/openapi.go]: Why comment this out?

File: openapi/openapi.go [hunk 6]
```diff
@@ -11,11 +11,27 @@ type Config struct {
...
+	Servers    []Server // TODO fix bugs after this modification
```
Comment (SasSwart) [inline openapi/openapi.go]: Is this TODO still relevant? I've been seeing it around for a while and its not clear enough for me to know what to go fix.

File: openapi/operation.go [hunk 7]
```diff
@@ -29,11 +30,17 @@ func (o *Operation) getChildren() map[string]Traversable {
 	if o == nil {
 		return children
 	}
+	// Parameters
 	for i := range o.Parameters {
 		parameter := o.Parameters[i]
-		children[string(i)] = parameter
+		children[string(rune(i))] = &parameter
```
Comment (SasSwart) [inline openapi/operation.go]: I don't think this is doing what you want it to do:
https://go.dev/play/p/Ti1cl_OQuTF

File: openapi/request_body.go [hunk 8]
```diff
@@ -1,14 +1,20 @@
...
+// RequestBody is a programmatic representation of the RequestBody object defined here: https://swagger.io/specification/#request-body-object
 type RequestBody struct {
 	node
+	Ref         string
 	Description string
 	Content     map[string]*MediaType
 	Required    bool
 }
 
+//func (r *RequestBody) GetName() string {
```
Comment (SasSwart) [inline openapi/request_body.go]: Still need this?

File: openapi/media_type.go (hunk 1 above)
Comment (alexthemayers) [reply openapi/media_type.go]: not entirely sure. Removed it

File: openapi/media_type_test.go (hunk 2 above)
Comment (alexthemayers) [reply openapi/media_type_test.go:27]: TODO'ed it inline

File: openapi/node.go (hunk 3 above)
Comment (alexthemayers) [reply openapi/node.go]: sorted. Adjusted the others to match.

File: openapi/node_test.go (hunk 4 above)
Comment (alexthemayers) [reply openapi/node_test.go:9]: Please elaborate? I'm dropping a todo here in the meantime

File: openapi/openapi.go (hunk 6 above)
Comment (alexthemayers) [reply openapi/openapi.go]: after the modifications made during hackday, it's fallen out of relevance

File: openapi/openapi.go (hunk 5 above)
Comment (alexthemayers) [reply openapi/openapi.go]: vestigial. removing

File: openapi/operation.go (hunk 7 above)
Comment (alexthemayers) [reply openapi/operation.go]: at the time I was just taking intellij's recommendation :|

File: openapi/request_body.go (hunk 8 above)
Comment (alexthemayers) [reply openapi/request_body.go]: ah yes. my bad

File: openapi/operation.go (hunk 7 above)
Comment (alexthemayers) [reply openapi/operation.go]: opted for `strconv.Itoa()` instead

File: openapi/node_test.go (hunk 4 above)
Comment (SasSwart) [reply openapi/node_test.go:9]: Having a PathItem does not mean having the right PathItem.
Its not too farfetched for a future bug in something like Dig to cause it to always return the first PathItem or for a dev to forget that they should be working with pointers and introduce a pass by value instead.

Not a biggie in this case, which is why its non blocking. I'm resolving the thread

**Pull request merged**